pkg/api/meta/meta.go

```diff
@@ -29,8 +30,15 @@ import (
 // obj must be a pointer to an API type. An error is returned if the minimum
 // required fields are missing. Fields that are not required return the default
 // value and are a no-op if set.
-// TODO: add a fast path for *TypeMeta and *ObjectMeta for internal objects
-func Accessor(obj interface{}) (Interface, error) {
+func Accessor(obj interface{}) (Object, error) {
+	if oi, ok := obj.(ObjectMetaAccessor); ok {
+		if om := oi.GetObjectMeta(); om != nil {
+			return om, nil
+		}
+	}
+	if oi, ok := obj.(Object); ok {
```

Comment:
> nit: Do we ever get to this branch? You would have to pass something that implements all those Object methods here, which doesn't seem to be a real usecase..
>  Add a line note

Test code maybe?

The lines below the diff seem unusual though.  Its for handling cases where I've explicitly (not anonymously) include `TypeMeta` and `ObjectMeta`?